### ISSUE TITLE
How do I switch the video address

### ISSUE THREAD

**Issue by a234146540** [brianegan/chewie#128]
How do I switch the video address?Can that big brother write an example?I am a super novice.

**Comment by cbenhagen** [comment]
You can check how the example is doing it:
https://github.com/brianegan/chewie/blob/master/example/lib/main.dart#L117-L128

**Comment by brianegan** [comment]
If that code is hard to follow, the general process is:

  1. Create a `ChewieController` with a `VideoPlayerController` that points to the original url
  2. Dispose of the controller when you want to change videos.
  3. Create a new `ChewieController` with a `VideoPlayerController` that points to the new url.

**Comment by a234146540** [comment]
``class _videoViewState extends State<videoView> {
  VideoPlayerController videoPlayController;
  ChewieController chewieController;
  Chewie playerWidget;
   void initState() {
    this.videoPlayController = VideoPlayerController.network('http://aaa.com/1.m3u8');
    this.chewieController = ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false,isLive: false);
    this.playerWidget = Chewie(
      controller: chewieController,
    );

}
  _test(){
this.chewieController.dispose();

                    this.videoPlayController.pause();
                    this.videoPlayController.seekTo(Duration(seconds: 0));
                   
                    this.videoPlayController=VideoPlayerController.network('http://aaa.com/2.mp4');

                    this.chewieController=ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false);
                    this.playerWidget=Chewie(controller: this.chewieController,);
}
}
Thank you very much for your reply.
Is that so?

**Comment by zfha** [comment]
> ``class _videoViewState extends State {
> VideoPlayerController videoPlayController;
> ChewieController chewieController;
> Chewie playerWidget;
> void initState() {
> this.videoPlayController = VideoPlayerController.network('http://aaa.com/1.m3u8');
> this.chewieController = ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false,isLive: false);
> this.playerWidget = Chewie(
> controller: chewieController,
> );
> 
> }
> _test(){
> this.chewieController.dispose();
> 
> ```
>                 this.videoPlayController.pause();
>                 this.videoPlayController.seekTo(Duration(seconds: 0));
>                
>                 this.videoPlayController=VideoPlayerController.network('http://aaa.com/2.mp4');
> 
>                 this.chewieController=ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false);
>                 this.playerWidget=Chewie(controller: this.chewieController,);
> ```
> }
> }
> Thank you very much for your reply.
> Is that so?

```
if (_controller != null) {
      _controller.pause();
      _controller.seekTo(Duration(seconds: 0));
    }
    if (_chewieController != null) {
      _chewieController.dispose();
    }
    _controller = new VideoPlayerController.network(videoUrl);
```
is right!

**Comment by PreviousTlx** [comment]
> > ``class _videoViewState extends State {
> > VideoPlayerController videoPlayController;
> > ChewieController chewieController;
> > Chewie playerWidget;
> > void initState() {
> > this.videoPlayController = VideoPlayerController.network('http://aaa.com/1.m3u8');
> > this.chewieController = ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false,isLive: false);
> > this.playerWidget = Chewie(
> > controller: chewieController,
> > );
> > }
> > _test(){
> > this.chewieController.dispose();
> > ```
> >                 this.videoPlayController.pause();
> >                 this.videoPlayController.seekTo(Duration(seconds: 0));
> >                
> >                 this.videoPlayController=VideoPlayerController.network('http://aaa.com/2.mp4');
> > 
> >                 this.chewieController=ChewieController(videoPlayerController: videoPlayController, aspectRatio: 16 / 9, autoPlay: true, looping: false);
> >                 this.playerWidget=Chewie(controller: this.chewieController,);
> > ```
> > 
> > 
> > }
> > }
> > Thank you very much for your reply.
> > Is that so?
> 
> ```
>     if (_controller != null) {
>       _controller.pause();
>       _controller.seekTo(Duration(seconds: 0));
>     }
>     if (_chewieController != null) {
>       _chewieController.dispose();
>     }
>     _controller = new VideoPlayerController.network(videoUrl);
>    _chewieController = ChewieController(
>       videoPlayerController: _controller,
>       autoPlay: true,
>       looping: true
>     );
> ```
> 
> is right!

It will be cause:
com.google.android.exoplayer2.ExoPlaybackException: com.google.android.exoplayer2.audio.AudioSink$InitializationException: AudioTrack init failed: 0, Config(44100, 12, 70880);
when you reload 15-20Times,How should i do?
